### PR TITLE
New javafx gui

### PR DESCRIPTION
**Pull request by Ljaysoft**
yet again...

### CODE REVIEW

**Comment by Wolfsblvt** [comment]
Reviewed and got the problem with IntelliJ tracked down now. The resources folder has to be in a very exact location.
Please create a folder named "main" inside the "src", then a folder called "resources" inside of that. Then it should work for both IDEs.

Files should be referenced in that way then:
`classLoader.getResource("layout/ChooseGUIWindow.fxml")`
Just relative to the resources folder.

And in that way, all files should be loaded. Even if you just need the path to them, use this loading system. For example for the icon, it should be this way:

```
URL image = classLoader.getResource("icon/PokeBall-icon.png");
primaryStage.getIcons().add(new Image(image.toExternalForm()));
```

As a last note: Add more and advanced exception handling if possible.
I got  a RuntimeException, with inner NullPointerException in `openGUIChooser`, which was unhandled.

**Comment by Ljaysoft** [comment]
Going to add a configuration option to show the new stuff only for developers, so the release won't show broken stuff

**Comment by Ljaysoft** [comment]
Changes made, needs code review and testing.

**Comment by wullxz** [comment]
Awesome! I hope I can test it this afternoon :)

**Comment by Ljaysoft** [comment]
Added a bunch of stuff

**Comment by Wolfsblvt** [comment]
Think we should not merge here until it's in a stable version, or we have fully planned out how to continue with this in dev.
We'll keep the PR open until then.

Just a note to every contributor here, so that they do not accidently merge this :P

**Comment by Ljaysoft** [comment]
Alright, I'll keep this Up to date when I can for the mean time.

At the same time, if my stuff isn't breaking anything on the old GUI, I'd really like to merge at some point just so I stop having conflicts with AccountController everytime you make a change @Wolfsblvt ;)

I spend more time merging stuff than writing code these past few days..

File: src/me/corriekay/pokegoutil/BlossomsPoGoManager.java [hunk 1]
```diff
@@ -1,21 +1,68 @@
...
 import me.corriekay.pokegoutil.controllers.AccountController;
+import me.corriekay.pokegoutil.utils.Config;
 import me.corriekay.pokegoutil.utils.Console;
 import me.corriekay.pokegoutil.utils.Utilities;
 
 
-public class BlossomsPoGoManager {
+public class BlossomsPoGoManager extends Application{
+	
+	public static final String VERSION = "0.2.0-Alpha";
```
Comment (Wolfsblvt) [inline src/me/corriekay/pokegoutil/BlossomsPoGoManager.java:22]: Can you please change it back to the current alpha? *lol*
Or better - that saves us one PR, change it to `v0.1.2-alpha.2`, which should be the next release. At least like planned now.

File: src/me/corriekay/pokegoutil/controllers/AccountController.java [hunk 2]
```diff
@@ -186,33 +201,146 @@ public static void logOn() throws Exception {
...
+			config.delete("login.PTCUsername");
+			config.delete("login.PTCPassword");
+			String authCode = config.getString("login.GoogleAuthToken", null);
+			boolean refresh = false;
+			if(authCode == null) {
+				//We need to get the auth code, as we do not have it yet.
+				UIManager.put("OptionPane.okButtonText", "Ok");
+				JOptionPane.showMessageDialog(null, "You will need to provide a google authentication key to log in. Press OK to continue.", "Google Auth", JOptionPane.PLAIN_MESSAGE);
+				//We're gonna try to load the page using the users browser 
+				if(Desktop.isDesktopSupported()) {
```
Comment (Wolfsblvt) [inline src/me/corriekay/pokegoutil/controllers/AccountController.java:261]: Seems like you copied the old code before my fix. You should use our own Desktop class now.
To see how to implement, take a look at the old login functionality.

(And on a sidenote, if that's the new login function, why are you using GUI elements in here? :D )

File: src/me/corriekay/pokegoutil/controllers/AccountController.java [hunk 3]
```diff
@@ -186,33 +201,146 @@ public static void logOn() throws Exception {
...
+	private static boolean checkSaveAuth2() {
+		return true;
+	}
+	
+	public static void logOff2() {
+		
+	}
+	
+	//TODO does nothing yet
+	public static void relogNewUser(){
```
Comment (Wolfsblvt) [inline src/me/corriekay/pokegoutil/controllers/AccountController.java:336]: That function was removed. We just keep logoff() and login(). Think you added it accidently again?

File: src/me/corriekay/pokegoutil/controllers/AccountController.java [hunk 4]
```diff
@@ -186,33 +201,146 @@ public static void logOn() throws Exception {
...
+		
+	}
 
     private static void alertFailedLogin(String message) {
```
Comment (Wolfsblvt) [inline src/me/corriekay/pokegoutil/controllers/AccountController.java:340]: That's a merge conflict again here. This is the new function, with the message inside. You should keep this and mark it as deprecated.

The old one above (that gets added again by you) should be removed.

On your own alertFailedLogin2() there should be the message parameter too then I would say. It's better to print the login fail message.

File: src/me/corriekay/pokegoutil/controllers/AccountController.java (hunk 3 above)
Comment (Ljaysoft) [inline src/me/corriekay/pokegoutil/controllers/AccountController.java:336]: the duplicates were so I have something to call from the UI, so I don't call UIManager things

File: src/me/corriekay/pokegoutil/controllers/AccountController.java (hunk 4 above)
Comment (Ljaysoft) [inline src/me/corriekay/pokegoutil/controllers/AccountController.java:340]: mm.. I'll look over the whole class then

File: src/me/corriekay/pokegoutil/BlossomsPoGoManager.java (hunk 1 above)
Comment (Ljaysoft) [inline src/me/corriekay/pokegoutil/BlossomsPoGoManager.java:22]: will change

**Comment by Wolfsblvt** [comment]
Okay, let's do it another way. I'll merge. Then I'll do the changes and you review shortly if it's okay.

After that, please as the next target, work on a bit cleanup of unused code and the doubled resource files. Second is even more important :P

**Comment by xephael** [comment]
@elliottfeng You should open a separate issue for that.
@Wolfsblvt I believe he means would it be possible to add a method of powering up a pokemon as much as possible (till you run out of candy/stardust or until it reaches max level)

**Pull request merged**